ceph: updated multus documentation with examples (#8021)
rook/rook: Documentation/ceph-cluster-crd.md

  name: rook-ceph
  namespace: rook-ceph
spec:

> # enabling network requires NetworkAttachmentDefinition `rook-public-nw` and `rook-cluster-nw` to be present in the `rook-ceph` namespace.

Reviewer: These high level examples really just focus on the minimal settings, with no commented settings. I still think the multus example does not need to be included in these high level examples.

Author: removed the commented settings

For more details, see the [Stretch Cluster design doc](https://github.com/rook/rook/blob/master/design/ceph/ceph-stretch-cluster.md).

> ## Cluster on Multus

Reviewer: These examples at the top of the doc are really focused on the different high level approaches to storage: raw devices, clusters on PVCs, and stretch clusters. I think the network example could apply to any of the storage examples, I think it's better to remove this high level example. We just need clarity in the network settings section.

Author: Can we put this in `Host-based Cluster` example?

> This example requires Network-attachment-definitions `rook-public-nw` and `rook-cluster-nw` to be present in the `rook-ceph` namespace.

Reviewer 2: ```suggestion
This example requires NetworkAttachmentDefinition `rook-public-nw` and `rook-cluster-nw` to be present in the `rook-ceph` namespace.
```

Based on the configuration, the operator will do the following:

  1. If only the `public` selector is specified, all communication will happen on that network
```yaml
  network:
    provider: multus
    selectors:
      public: rook-ceph/rook-public-nw
```
  2. If only the `cluster` selector is specified, the internal cluster traffic* will happen on that network. All other traffic to mons, OSDs, and other daemons will be on the default network.

> * Internal cluster traffic includes OSD heartbeats, data replication, and data recovery

Reviewer: I'd suggest moving this * below the example of bullet 3. It is also rendering as a bullet instead of a note, so perhaps it needs to be escaped to remain as a *

Author: done

```yaml
  network:
    provider: multus
    selectors:
      cluster: rook-ceph/rook-cluster-nw
```

>   3. If both `public` and `cluster` selectors are specified the first one will run all the communication network and the second the internal cluster network

Reviewer: A * for the note 
```suggestion
  3. If both `public` and `cluster` selectors are specified the first one will run all the communication network and the second the internal cluster network*
```

Based on the configuration, the operator will do the following:

>   1. If only the `public` selector is specified communication will happen on that network

Reviewer: All communication? 
```suggestion
  1. If only the `public` selector is specified, all communication will happen on that network
```

Author: Yes, all communication.

```yaml
  network:
    provider: multus
    selectors:
      public: rook-ceph/rook-public-nw
```

>   2. If only the `cluster` selector is specified OSD communication and replication will happen on that network.

Author: What about the following?

If only the `cluster` selector is specified, OSD heartbeat, object replication and recovery will happen on that network

>   2. If only the `cluster` selector is specified OSD heartbeat, object replication and recovery will happen on that network.

Reviewer: Where is all the public traffic? On the default CNI, right? Also it would be good to define the internal traffic as a separate note so that bullet 3 can also reference it
```suggestion
  2. If only the `cluster` selector is specified, the internal cluster traffic* will happen on that network. All other traffic to mons, OSDs, and other daemons will be on the default network.

* Internal cluster traffic includes OSD heartbeats, data replication, and data recovery
* ```

Author: updated

```yaml
  network:
    provider: multus
    selectors:
      cluster: rook-ceph/rook-cluster-nw
```

>   3. If both `public` and `cluster` selectors are specified the first one will run the communication network and the second the replication network

Reviewer: The internal cluster network would refer to the same note as above
```suggestion
  3. If both `public` and `cluster` selectors are specified the first one will run the public communication network and the second the internal cluster network*
```

```

* Ensure that `master` matches the network interface of the host that you want to use.
* Ipam type `whereabouts` is required because it makes sure that all the pods get a unique IP address from the multus network.

> * The NAD should be referenced along with the namespace in which it is present like `public: <namespace>/<name of NAD>`.

Reviewer 2: ```suggestion
* The NetworkAttachmentDefinition should be referenced along with the namespace in which it is present like `public: <namespace>/<name of NAD>`.
```

  e.g., the network attachment definition are in `default` namespace:
  ```yaml
    public: default/rook-public-nw
    cluster: default/rook-cluster-nw
  ```

>   * This format is required in order to use the NAD across namespaces.

Reviewer 2: ```suggestion
  * This format is required in order to use the NetworkAttachmentDefinition across namespaces.
```

>   * In Openshift, to use the NetworkAttachmentDefinition across namespaces, the NAD must be deployed in the default namespace and it can be referenced as `default/myNAD` where `default` is the namespace and `myNAD` is the network attachment definition.

Reviewer 3: Let's rewrite this to be more clear.
In Openshift, to use a NetworkAttachmentDefinition (NAD) across namespaces, the NAD must be deployed in the `default` namespace. The NAD is then referenced with the namespace: `default/rook-public-nw`

Author: Made the changes. Thanks
